### ISSUE TITLE
GET API taskList vs events

### ISSUE THREAD

**Issue by ffazzini** [Activiti/Activiti#1705]
To return a list of objects in the body response.

Can we use the same strategy? Either we call both *List
or just events and tasks.

I would go for the latter.

**Comment by Salaboy** [comment]
@ffazzini please remind me to have a quick discussion about this one. I'm not sure that I understand what the problem is from the description of the issue.

**Comment by ffazzini** [comment]
Sure sorry, basically get Tasks endpoint returns: 
```javascript
{
    "_embedded": {
        "taskList": [
            {
                "id": "4a86c33b-02b2-11e8-8ac0-0a5864600118",

while get Events:
```javascript
{
    "_embedded": {
        "events": [
            {

taskList vs events

**Comment by Salaboy** [comment]
@ryandawsonuk @erdemedeiros @ffazzini I think that we should close this one, it might confuse some other people. tasks, processes, and variables are not events, and events are what is expected to get from audit. 
Are we all ok with closing this?

**Comment by ffazzini** [comment]
For closing you mean fixing it? It is good for clients having a consistent way to parse lists.

**Comment by ryandawsonuk** [comment]
You mean whether 'List' is appended at the end? I'd say we ask @mauriziovitale . My impression from https://github.com/Activiti/Activiti/issues/1745 is that using camelcase over hyphen matters but not sure about whether 'List' needs to be at the end of very list.

**Comment by ffazzini** [comment]
yes @ryandawsonuk , sorry if that was unclear above

**Comment by mauriziovitale** [comment]
@ffazzini @Salaboy @ryandawsonuk 
For my point of view, we should have a generic response structure like 
{
    "_embedded": {
        "items": [

In this way, we can have a single Response model that accepts a specific type like TaskList or EventList         or whatever

**Comment by erdemedeiros** [comment]
Hi Guys,
Concerning the `List` suffix it is probably something similar to the [RelProvider](https://docs.spring.io/spring-hateoas/docs/current/reference/html/#spis.rel-provider). Maybe we have a different configuration for events. Or we have Evo Inflector in the classpath, for some reason. I agree that's better to have the same pattern everywhere.

Concerning structure if we follow the Alfresco guidelines (https://github.com/Activiti/Activiti/issues/1419) it will be something like
```
{
    "list": {
        "entries": [
            {
                "id": "SimpleProcess:2:0f0e1a0c-0b58-11e8-a8ed-de0eaec1cca0",
                "name": "SimpleProcess",
                "version": 2
            }
        ],
        "pagination": {
            "skipCount": 3,
            "maxItems": 3,
            "count": 1,
            "hasMoreItems": false,
            "totalItems": 4
        }
    }
}
```

**Comment by ffazzini** [comment]
true @erdemedeiros  i think you also have an extra 'entry' field inside 'entries'

**Comment by mteodori** [comment]
@ffazzini @erdemedeiros I can confirm, it should be:

```json
    "list": {
        "entries": [
            {
                "entry": {
                    "id": "SimpleProcess:2:0f0e1a0c-0b58-11e8-a8ed-de0eaec1cca0",
                    "name": "SimpleProcess",
                    "version": 2
                }
            }
        ],
        "pagination": {
            "skipCount": 3,
            "maxItems": 3,
            "count": 1,
            "hasMoreItems": false,
            "totalItems": 4
        }
    }
}
```

**Comment by erdemedeiros** [comment]
@ffazzini @mteodori thanks. I'll update it.

**Comment by Salaboy** [comment]
@erdemedeiros good so this is going to be fixed here: https://github.com/Activiti/Activiti/issues/1419 can we close this one then?

**Comment by Salaboy** [comment]
I'm closing this one because this is going to be fixed in #1419
